### PR TITLE
api_specification should use the platform name it was supplied

### LINKED ISSUE #1120: Installing the pry gem on Windows fails starting with rubygems 2.4.5
```
gem update --system 2.4.5
gem install pry
ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
    bad response Not Found 404 (https://bb-m.rubygems.org/quick/Marshal.4.8/pry-0.10.1-x86-mingw32.gemspec.rz)

gem update ---system 2.4.4
gem install pry
Successfully installed pry-0.10.1-x86-mingw32
Parsing documentation for pry-0.10.1-x86-mingw32
1 gem installed
```

### PR DESCRIPTION
**Pull request by jaym**
Without this patch, `gem install pry` fails because it wants to download pry-0.10.1-x86-mingw32 
instead of pry-0.10.1-i386-mingw32. This is because a Platform object is created from the 
`i386-mingw32` string, which gets normalized into `x86-mingw32`. This behavior was
introduced in d96c57ce3c69dbaf3c9bb43e65eaa2c5e2601718.

This fixes #1120.

### CODE REVIEW

**Comment by tenderlove** [comment]
Any way we can test this?

**Comment by jdmundrawala** [comment]
@tenderlove Modified the existing spec to test this case. I'm not sure if https://github.com/rubygems/rubygems/pull/1184/files#diff-517c8e9597aa64b456816c0f220e12deR41 is a bug... it tripped me up though

**Comment by tenderlove** [comment]
@drbrain or @evanphx can you check this?  I think it will fix #1120, #1219, and #1152.  If I'm reading these correctly, platform specific gems can't be installed on Windows (and that's most native gems AFAICT).  It seems like a deal breaker as far as upgrading is concerned for windows folks (which is bad news considering [this](http://blog.rubygems.org/2015/06/08/2.4.8-released.html))

File: test/rubygems/test_gem_resolver_api_specification.rb [hunk 1]
```diff
@@ -31,13 +31,14 @@ def test_initialize
   def test_fetch_development_dependencies
     specs = spec_fetcher do |fetcher|
       fetcher.spec 'rails', '3.0.3' do |s|
+        s.platform = 'i386-mingw32'
```
Comment (zzak) [inline test/rubygems/test_gem_resolver_api_specification.rb]: Should we have another test which modifies this so we don't change coverage of the original tests? i.e. add a separate test that does this

**Comment by evanphx** [comment]
Why would we not want normalization?

**Comment by evanphx** [comment]
I don't see this as a valid fix, especially since this change was introduce do fix a different bug where normalization was wanted. So this reverts that which will regress the other bug.

**Comment by tenderlove** [comment]
> Why would we not want normalization?

I really don't know (which is why I'm roping you in).  @drbrain [said that we don't want normalization](https://github.com/rubygems/rubygems/issues/1120#issuecomment-91374806), but I'm not sure why.

**Comment by jdmundrawala** [comment]
I think you dont want normailzation in this specific case because then https://github.com/rubygems/rubygems/blob/d3db595be39639fdc6b020e24fab4d0cf052b448/lib/rubygems/name_tuple.rb#L76 would be incorrect. It is useful in other cases. I'm not sure this regresses the other bug as there was a test that still passes.

**Comment by jaym** [comment]
Any thoughts on what the right solution is for this?

**Comment by jaym** [comment]
https://github.com/jaym/rubygems/commit/225e3b62691034073f84c8e7dc1381b714710160 probably won't regress the bug fix

**Comment by indirect** [comment]
@jaym can you summarize the current state of this PR?

**Comment by jaym** [comment]
I'm guessing it's not really affecting anyone anymore with the change to pry. I'll go ahead and close it
